### ISSUE TITLE
Feature: Refreshing output (or switching terminal buffer) window without closing it

### ISSUE THREAD

**Issue by wookayin** [nvim-neotest/neotest#170]
TL;DR) I wish there were a feature to refresh the Neotest Output window (e.g., `neotest.output.open()`) with the requested (or the current) test case, when the output or output_panel window is already open.

Currently, the existing window (showing an output for another test case the window was opened for) would remain unchanged, when another `:lua neotest.output.open()` call was made for some other test cases. To help your understanding, I'm attaching a demo video:

![DEMO: Output not refreshing](https://user-images.githubusercontent.com/1009873/203451066-9390f5e3-aed7-4678-9d00-d419615a69c5.gif)

- 1. Execute `test_bar` and open an output window (shown below)
- 2. Execute `test_foo` and request `neotest.output.open` for `test_foo` (either automatically upon failing or manually), but the existing output window -- either `attach` or `output` -- won't refresh its content.
- 3. Close the neotest output window, and request `output.open` for `test_foo` again. Now it shows the output for `test_foo`.

This feature will be useful when one needs to have a terminal open side-by-side and iterate on modifying code and running tests very frequently. Having the output shown in a floating window would be, in this particular use case, a bit difficult to follow.

One can quit an existing neotest output window (if any) and open it again, but this may not be sufficient for my use case because I would like to retain the window (its position, size, etc.).

So it would be more like changing/switching the terminal buffer displayed on the neotest window when `neotest.output.open()` is called. How can we have this feature implemented?

**Comment by rcarriga** [comment]
This sounds more like you want to use the new `output_panel` consumer, would that suit your need more?

**Comment by wookayin** [comment]
Hmm interesting, indeed output panel can update the output when another test runs:

![output_panel](https://user-images.githubusercontent.com/1009873/203629492-38859f82-f44d-4995-8d90-2bffa36b3773.gif)

This will be more helpful for my use case here, but one issue is that it doesn't *stream* the output (like `neotest.run.attach()` does). The description of `output_panel` reads `A consumer that streams all output of tests to a terminal window`, so I would expect this would stream all the interim output before the test finishes also printed out as soon as the buffer flushes. This somehow sounds overlapping with `run.attach` a bit, but would that be possible somehow?

**Comment by rcarriga** [comment]
Unfortunately not, streaming outputs directly from the processes would have too many pitfalls. See https://github.com/nvim-neotest/neotest/issues/123#issuecomment-1316574378 for reasons

**Comment by rcarriga** [comment]
I'll add a side note if you really wanted this feature you could write a custom strategy that would be similar to the integrated one but would just reuse the same buffer/window and keep them persistent. This wouldn't work for the general case but it should be completely workable for python and if you don't run tests concurrently.
